honeybee/boundarycondition.py

```diff
@@ -64,57 +120,73 @@ def wind_exposure_idf(self):
-            }
-        if hasattr(self, 'boundary_condition_object'):  # Surface boundary condition
-            bc_dict['bc_object'] = self.boundary_condition_object
+            bc_dict['sun_exposure'] = self.sun_exposure
+            bc_dict['wind_exposure'] = self.wind_exposure
+            bc_dict['view_factor'] = self.view_factor
         return bc_dict
 
-    def ToString(self):
-        return self.__repr__()
-
-    def __repr__(self):
-        return self.name
-
 
 class Surface(_BoundaryCondition):
     """Bondary condition when an object is adjacent to another object."""
 
     __slots__ = ('_boundary_condition_objects',)
 
-    def __init__(self, other_object):
+    def __init__(self, boundary_condition_objects):
         """Initialize Surface boundary condition.
 
+        Args:
+            boundary_condition_objects: A list of up to 3 object names that are adjacent
+                to this one. The first object is always the one that is immediately
+                adjacet and is of the same object type (Face, Aperture, Door). When
+                this boundary condition is applied to a Face, the second object in the
+                tuple will be the parent Room of the adjacent object. When the boundary
+                condition is allplied to a sub-face (Door or Aperture), the second object
+                will be the parent Face of the sub-face and the third object will be the
+                parent Room of the adjacent sub-face.
+        """
+        try:
+            self._boundary_condition_objects = tuple(str(v) for v in boundary_condition_objects)
+        except (ValueError, TypeError):
+            raise TypeError('Input boundary_condition_objects must be a list of strings.')
+        assert 2 <= len(boundary_condition_objects) <= 3, \
+            'Input boundary_condition_objects length must be 2 or 3 not {}'.format(
+                len(boundary_condition_objects))
+
+    @classmethod
+    def from_dict(cls, data):
+        """Initialize BoundaryCondition from a dictionary.
+
+        Args:
+            data: A dictionary representaion of the boundary condition.
+        """
+        return cls(data['boundary_condition_objects'])
+
+    @classmethod
+    def from_other_object(cls, other_object):
+        """Initialize Surface boundary condition from an adjacent other object.
+
         Args:
             other_object: Another object (Face, Aperture, Door) of the same type
                 that this boundary condition is assigned.  This other_object will be
                 set as the adjacent object in this boundary condition.
         """
-        _BoundaryCondition.__init__(self)
-        self._boundary_condition_objects = []
+        _boundary_condition_objects = []
         if other_object is not None:
-            self._boundary_condition_objects.append(other_object.name)
+            _boundary_condition_objects.append(other_object.name)
             if other_object.has_parent:
```

Comment:
shouldn't we raise an Exception here? Are you allowing Surface boundary condition for a standalone object?

Reply:
I think you are right that I have been doing this check in the wrong place.  Right now, I have been doing it in the `set_adjacency` methods on the objects rather than here in the boundary condition.  I originally thought this made sense because, for Apertures, you need to check 2 levels of parents while, for Face, you only need to check 1.  I think I can build it into here by adding an option to include the number of levels to check.

Reply:
Done. I think and I think this is better.  People at least get more informative error messages when they set the adjacency for an object that does not have a parent Room.

Reply:
Cool!